controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -165,57 +194,10 @@ func (r *StorageConsumerReconciler) reconcileEnabledPhases() (reconcile.Result,
 			return reconcile.Result{}, err
 		}
 
-		if _, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, consumerConfigMap, func() error {
-			if consumerConfigMap.Data == nil {
-				consumerConfigMap.Data = map[string]string{}
-			}
-
-			defaultConsumerResourceNames := util.GetStorageConsumerDefaultResourceNames(
-				r.storageConsumer.Name,
-				string(r.storageConsumer.UID),
-			)
-			for key := range defaultConsumerResourceNames {
-				consumerConfigMap.Data[key] = cmp.Or(
-					strings.Trim(consumerConfigMap.Data[key], " "),
-					defaultConsumerResourceNames[key],
-				)
-			}
-
-			// Get config map's controller reference
-			controllerIndex := slices.IndexFunc(
-				consumerConfigMap.OwnerReferences,
-				func(ref metav1.OwnerReference) bool { return ptr.Deref(ref.Controller, false) },
-			)
-			var controllerRef *metav1.OwnerReference
-			if controllerIndex != -1 {
-				controllerRef = &consumerConfigMap.OwnerReferences[controllerIndex]
-			}
-			// If there is no controller ref, take control over the config map
-			if controllerRef == nil {
-				if err := controllerutil.SetControllerReference(
-					r.storageConsumer,
-					consumerConfigMap,
-					r.Scheme,
-				); err != nil {
-					return err
-				}
-				// If I am not the config map controller add me as an owner
-			} else if controllerRef.UID != r.storageConsumer.UID {
-				if err := controllerutil.SetOwnerReference(
-					r.storageConsumer,
-					consumerConfigMap,
-					r.Scheme,
-				); err != nil {
-					return err
-				}
-			}
-			return nil
-		}); err != nil {
+		if err := r.reconcileConsumerConfigMap(consumerConfigMap, isRbdAvailable, isCephFsAvailable, isNfsAvailable); err != nil {
```

Comment:
You already have 3 variables for  3 services but we also need to take into account noobaa, and maybe other services in the future. To make sure the code is not cluttered i would suggest using a new struct `AvailableServices`